packages/azure-http-specs/specs/azure/client-generator-core/client-initialization/main.tsp

```diff
@@ -0,0 +1,120 @@
+import "@typespec/http";
+import "@typespec/spector";
+import "@azure-tools/typespec-client-generator-core";
+
+using Http;
+using Spector;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for client initialization decorator - moving parameters from method to client level")
+@scenarioService("/azure/client-generator-core/client-initialization")
+namespace Service;
+
+// Common parameter types and models
+
+model NameHeaderParam {
+  @header("name")
+  name: string;
+}
+
+model RegionQueryParam {
+  @query
+  region: string;
+}
+
+model Input {
+  name: string;
+}
+
+// Scenario 1: Header parameter moved to client level
+
+@route("/header-param")
+interface HeaderParam {
+  @route("/with-query")
+  @get
+  withQuery(...NameHeaderParam, @query id: string): string;
```

Comment:
For any of these `string` in response, if we intend it as JSON string (appears so in mockapi), we need to write a `@header contentType: "application/json"` now.
(found out when I try generate the code...)

Otherwise, they be interpretated as `application/text`.

see https://typespec.io/docs/release-notes/release-2025-03-18/

Reply:
Thanks for this catch @weidongxu-microsoft! Let me just switch it to `text/plain` responses